src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -841,8 +841,12 @@ public bool Equals(Guid g)
             // Now compare each of the elements
             return g._a == _a &&
                 Unsafe.Add(ref g._a, 1) == Unsafe.Add(ref _a, 1) &&
+#if BIT64
+                Unsafe.Add(ref Unsafe.As<int, ulong>(ref g._a), 1) == Unsafe.Add(ref Unsafe.As<int, ulong>(ref _a), 1);
```

Comment:
This has potential portability problem: It assumes that accessing misaligned ulong is ok.

Reply:
I agree 🙁

Reply:
So this should rather use `Unsafe.ReadUnaligned`